types/types.go

```diff
@@ -160,3 +161,47 @@ func (m ComMatrix) Contains(cd ComDetails) bool {
 
 	return false
 }
+
+func ToNFTables(m ComMatrix) ([]byte, error) {
+	var tcpPorts []string
+	var udpPorts []string
+	for _, line := range m.Matrix {
+		if line.Protocol == "TCP" {
+			tcpPorts = append(tcpPorts, fmt.Sprint(line.Port))
+		} else if line.Protocol == "UDP" {
+			udpPorts = append(udpPorts, fmt.Sprint(line.Port))
+		}
+	}
+
+	tcpPortsStr := strings.Join(tcpPorts, ", ")
+	udpPortsStr := strings.Join(udpPorts, ", ")
+
+	result := fmt.Sprintf(`#!/usr/sbin/nft -f
+
+	table ip filter {
+		chain OPENSHIFT {
+			# Allow loopback traffic
+			iif lo accept
+	
+			# Allow established and related traffic
+			ct state established,related accept
+	
+			# Allow ICMP
+			ip protocol icmp accept
+
+			# Allow specific TCP and UDP ports
+			tcp dport  { %s } accept
+			udp dport { %s } accept
+
+			# Logging and default drop
+			log prefix "firewall " drop
+		}
+	
+		chain INPUT {
+			type filter hook input priority 0; policy accept;
+			jump FIREWALL
```

Comment:
Why not just insert all the rules in the OPENSHIFT chain here and skip the chain jump?

Reply:
because the input chain might already exist and have other stuff in it
or at least that was my initial thought when I saw this - that we have everything openshift need in it's own chain, and that can be used anywhere the user wants..
but then we might need a different way to modify the input chain?

Reply:
I know. The thing is another entity manages the whole table so we should not touch it. There is even a warning comment in it. That is another advantage of nftables compared to iptables: you can have multiple tables. Also important is if two tables have chains on the same Netfilter hook, it is paramount to have them with different priorities or otherwise there is no guarantee of evaluation order:

> It's possible to give two base chains the same priority, but there is no guaranteed evaluation order of base chains with identical priority that are attached to the same hook location

https://wiki.nftables.org/wiki-nftables/index.php/Configuring_chains

In the latest PR change, Amal addressed my comment about the priority and table separation where the communication matrix rules have lower priority than other ones. If it had higher or same but nftables would evaluate those rules first, traffic could be dropped (there is a drop-all rule with log at the end of the proposed chain).

Reply:
Yuval, think of firewalld. It creates its own table so that it doesn't conflict with anything else on the system.